dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java

```diff
@@ -243,6 +243,31 @@ private List<ObjectNode> buildDataxJobContentJson() {
                 dataxTaskExecutionContext.getTargetType(),
                 dataxTaskExecutionContext.getTargetConnectionParams());
 
+        ObjectNode reader;
+        ObjectNode writer;
+
+        switch (dataxTaskExecutionContext.getSourcetype()) {
+            default:
+                reader = buildDataxJobContentJsonDefaultReader(dataSourceCfg);
+        }
+
+        switch (dataxTaskExecutionContext.getTargetType()) {
+            case DORIS:
+            default:
+                writer = buildDataxJobContentJsonDefaultWrite(dataSourceCfg, dataTargetCfg);
+        }
+
+        List<ObjectNode> contentList = new ArrayList<>();
+        ObjectNode content = JSONUtils.createObjectNode();
+        content.set("reader", reader);
+        content.set("writer", writer);
+        contentList.add(content);
+
+        return contentList;
+    }
+
+    private ObjectNode buildDataxJobContentJsonDefaultReader(BaseConnectionParam dataSourceCfg) {
```

Comment:
Could you move this from DataXTask? and use a new interface to generate datax file content?

Reply:
ASAP